src/fu-plugin.c

```diff
@@ -510,6 +511,15 @@ fu_plugin_device_remove (FuPlugin *plugin, FuDevice *device)
 		return;
 	}
 
+	children = fu_device_get_children (device);
+	for (guint j = 0; j < children->len; j++) {
+		FuDevice *child = g_ptr_array_index (children, j);
+		g_debug ("emit removed from %s: %s",
+			fu_plugin_get_name (plugin),
+			fu_device_get_id (child));
+		g_signal_emit (plugin, signals[SIGNAL_DEVICE_REMOVED], 0, child);
```

Comment:
I'm not sure about this. Maybe there are some logical problems in removing a device that the plugin still thinks is "active"? I'd feel a lot more warm and fuzzy with some self tests.

Reply:
Added a self test for it.

In my mind I think that's why the plugins also need to get the `usb_device_removed` or `udev_device_removed` call back.

Reply:
The thing is, that's a signal that says "this hardware has been removed", not "the daemon has killed this object for you". I guess my concern could be mitigated by adding some comments to the fu_device_add_parent_guid function that says that when the parent is removed the child is also going to be removed. I think for most plugins that don't store a ref of the FuDevice in the plugin private data it's fine.

Reply:
You know I think actually the Dell plugin keeps refs and needs to be updated for this flow.  I'll add a comment to that function and fixup the Dell plugin.